processing/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -142,6 +142,26 @@ public static void serialize(ByteBuffer buffer, int position, ExpressionType typ
     }
   }
 
+  public static byte[] toBytes(ExpressionType expressionType, NullableTypeStrategy<Object> strategy, Object o)
+  {
+    // convert the array to byte[] form so that we take a hash of the whole array
+    final ExprEval<?> eval = ExprEval.ofType(expressionType, o);
+    final int size = strategy.estimateSizeBytes(eval.valueOrDefault());
+    final ByteBuffer buffer = ByteBuffer.allocate(size);
+    strategy.write(buffer, eval.valueOrDefault(), size);
+    return buffer.array();
+  }
+
+  public static byte[] toBytesBestEffort(Object o)
+  {
+    final ExprEval<?> eval = ExprEval.bestEffortOf(o);
+    final NullableTypeStrategy<Object> strategy = eval.type().getNullableStrategy();
+    final int size = strategy.estimateSizeBytes(eval.valueOrDefault());
+    final ByteBuffer buffer = ByteBuffer.allocate(size);
+    strategy.write(buffer, eval.valueOrDefault(), size);
+    return buffer.array();
```

Comment:
I find myself feeling like these 4 lines could be reused by the other `toBytes()` as well.